src/backend/dx12/src/device.rs

```diff
@@ -2656,24 +2648,45 @@ impl d::Device<B> for Device {
                 .usage
                 .intersects(image::Usage::SAMPLED | image::Usage::INPUT_ATTACHMENT)
             {
-                self.view_image_as_shader_resource(info.clone()).ok()
+                let info = if info.range.aspects.contains(format::Aspects::DEPTH) {
+                    conv::map_format_shader_depth(surface_format)
+                        .map(|format| ViewInfo {
+                            format,
+                            .. info.clone()
+                        })
+                } else if info.range.aspects.contains(format::Aspects::STENCIL) {
+                    let new_swizzle = conv::swizzle_rg(swizzle);
```

Comment:
could you add a comment why this is needed? I guess dx12 maps stencil to the R channel here compared to Vulkan?

Reply:
it's the opposite: dx12 maps it to G, while Vulkan expects R

Reply:
anyway, added a small comment now